includes/OutputPage.php

```diff
@@ -2105,9 +2105,7 @@ public function output() {
 				$this->mRedirect = $this->getRequest()->getFullRequestURL();
 			}
 
-			if ( $this->mRedirectProtocol === PROTO_HTTP ) {
-				$this->mRedirect = wfHttpsToHttp( $this->mRedirect );
-			} elseif ( $this->mRedirectProtocol === PROTO_HTTPS ) {
+			if( wfHttpsAllowedForURL( $this->mRedirect ) ) {
```

Comment:
I'm not sure if that maintains the previous logic. Will have to double-check that later.

Reply:
`if ( $this->mRedirectProtocol === PROTO_HTTPS ) {` is more compatible with the previous flow. But while we're here, I think the `wfHttpsAllowedForURL` should be called after `wfExpandUrl`. So the flow should be:
```
# Standards require redirect URLs to be absolute
$this->mRedirect = wfExpandUrl( $this->mRedirect, $this->mRedirectProtocol );

if ( $this->mRedirectProtocol === PROTO_HTTPS ) {
    $this->mRedirect = wfHttpToHttps( $this->mRedirect );
}
```